lms/djangoapps/teams/views.py

```diff
@@ -485,8 +487,20 @@ def has_object_permission(self, request, view, obj):
             """Returns true if the user is enrolled or is staff."""
             return has_team_api_access(request.user, obj.course_id)
 
+    class IsStaffOrTAOrReadOnly(IsStaffOrReadOnly):
+        """Permission that checks to see if the user is global staff, course
+        staff, or has discussion privileges. If none of those conditions are
+        met, only read access will be granted.
+        """
+
+        def has_object_permission(self, request, view, obj):
+            return (
+                has_discussion_privileges(request.user, obj.course_id) or
+                IsStaffOrReadOnly.has_object_permission(self, request, view, obj)
```

Comment:
Let me experiment with this tomorrow when I have my work computer. I don't understand why it would be different, but perhaps the syntax I suggested is wrong.